scripts/common/repository.js

```diff
@@ -112,46 +110,107 @@ class Repository {
     });
   }
 
+  static fetchTags() {
+    return execa.shell('git describe --abbrev=0').then(({ stdout }) => stdout).catch(() => '');
+  }
+
+  createSmokeTag(tag) {
+    return execa.shell(`git tag ${this.name}@${this.version} ${tag}`);
+  }
+
+  deleteSmokeTag() {
+    return execa.shell(`git tag -d ${this.name}@${this.version}`);
+  }
+
   updateChangelog() {
-    const fetchTags = () => execa.shell('git describe --abbrev=0').then(({ stdout }) => stdout).catch(() => '');
-    const createSmokeTag = tag => execa.shell(`git tag ${this.name}@${this.version} ${tag}`);
-    const deleteSmokeTag = () => execa.shell(`git tag -d ${this.name}@${this.version}`);
     const getChangelog = () => execa.shell(`lerna exec --scope ${this.name} --concurrency 1 --no-sort -- conventional-changelog \
       --preset angular --lerna-package ${this.name} --commit-path ${this.location}`);
     const updateChangelog = () => execa.shell(`lerna exec --scope ${this.name} --concurrency 1 --no-sort -- conventional-changelog \
       --preset angular --infile CHANGELOG.md --same-file --lerna-package ${this.name} --commit-path ${this.location}`);
-    return fetchTags()
-      .then(createSmokeTag)
+    return this.constructor.fetchTags()
+      .then((...args) => this.createSmokeTag(args))
       .then(() => getChangelog()
         .then(({ stdout }) => {
           this.changelog = stdout.replace(/#/, `## ${this.name}`);
         }))
       .then(updateChangelog)
-      .then(deleteSmokeTag)
+      .then(() => this.deleteSmokeTag())
```

Comment:
```suggestion
      .then(this.deleteSmokeTag)
```

Reply:
No, this will be different with your suggestion :(

It's why i use an arrow function here :)

Reply:
Noticed. Thanks.